libyelp/yelp-uri.c

```diff
@@ -244,6 +244,7 @@ yelp_uri_resolve (YelpUri *uri)
         g_signal_connect_swapped (priv->res_base, "resolved",
                                   G_CALLBACK (resolve_start),
                                   uri);
+        g_object_ref (uri);
```

Comment:
Yeah, I think so, yes. [I didn't like the fact that we didn't add any `g_object_unref` to match this](https://github.com/kinvolk/yelp/pull/1#discussion_r121357036). Now I think that in the `g_signal_connect_swapped` above we need to add our own function that calls `resolve_start` and then unrefs the object.